pydantic/json_schema.py

```diff
@@ -35,11 +35,20 @@
 
     from . import ConfigDict
     from ._internal._dataclasses import PydanticDataclass
+    from .fields import FieldInfo
     from .main import BaseModel
 
 JsonSchemaValue = Dict[str, Any]
 
 
+def get_field_info(
+    schema: core_schema.CoreSchema | core_schema.TypedDictField | core_schema.DataclassField,
+) -> FieldInfo | None:
+    from pydantic._internal._generate_schema import METADATA_FIELD_INFO_KEY
+
+    return schema.get('metadata', {}).get(METADATA_FIELD_INFO_KEY, None)
```

Comment:
I would look at the `pydantic._internal._core_metadata.CoreMetadataHandler` which handles some other edge cases this doesn't (e.g., metadata exists and isn't a dict)